WT-15718 Favour internal SQLite3 library during build

ablekhman opened the pull request:
- Add `ENABLE_PALITE` flag to control whether PALite is included in the build (default=ON)
- Set default preference for SQLite3 library: `ENABLE_INTERNAL_SQLITE3=ON`

ablekhman committed:
WT-15718 Favour internal SQLite3 library during build
- Add `ENABLE_PALITE` flag to control whether PALite is included in the build (default=ON)
- Set default preference for SQLite3 library: `ENABLE_INTERNAL_SQLITE3=ON`

ivankochin commented on cmake/configs/base.cmake:
```diff
@@ -359,17 +364,10 @@ config_string(
     ENABLE_INTERNAL_SQLITE3
     "Enable internal SQLite3 library. If disabled, the system SQLite3 library will be used."
-    DEFAULT ${default_internal_sqlite3}
+    DEFAULT ON
```
Shouldn't this config depend on ENABLE_PALITE? i.e. why do we need SQLite if we don't use PALite?

ablekhman replied:
This flag only controls if internal or external sqlite3 library is used. If ENABLE_PALITE=OFF, then I simply skip the library discovery altogether in sqlite3.cmake

ivankochin commented on cmake/configs/base.cmake:
```diff
@@ -359,17 +364,10 @@ config_string(
-endif()
-
 config_bool(
     ENABLE_INTERNAL_SQLITE3
```
Can we make this config with 3 options: OFF, INTERNAL, SYSTEM?

ablekhman replied:
Let's leave it simple for now. It will require much bigger rewrite of the current cmake scripts.

ablekhman committed:
WT-15718 Favour internal SQLite3 library during build
- Add `ENABLE_PALITE` flag to control whether PALite is included in the build (default=ON)
- Set default preference for SQLite3 library: `USE_SYSTEM_SQLITE3=OFF`

ivankochin commented on cmake/third_party/sqlite3.cmake:
```diff
@@ -1,3 +1,7 @@
+endif()
+
 if(NOT ENABLE_INTERNAL_SQLITE3)
     find_package(SQLite3 ${SQLITE3_REQUIRED_VERSION} REQUIRED)
```
Can we just specify custom PATH and NO_DEFAULT_PATH options for find_package to find the internal one and avoid having all this regex parsing here?

Maybe it would require creating an additional cmake module file in the `${CMAKE_SOURCE_DIR}/test/3rdparty/sqlite3` folder.

```
find_package(SQLite3 REQUIRED
    PATHS "${CMAKE_SOURCE_DIR}/test/3rdparty/sqlite3"
    NO_DEFAULT_PATH)
```

ablekhman replied:
We cannot use `find_package` because we do not have proper sqlite3 distribution: header files, binary libs, etc. The "discovery" via regex matching in the bundled sources tries to look like the `find_package` call. It just for the convenience of the user, so the output looks more or less similar between using system library or the bundled sources.
